ably/rest_http_error.go

```diff
@@ -15,6 +15,7 @@ type RestHttpError struct {
 func NewRestHttpError(response *http.Response, message string) *RestHttpError {
 	err := &RestHttpError{Response: response, Msg: message}
 	err.readBody()
+	err.Response = nil
```

Comment:
It should, I need to rewrite RestHttpError either way as it's a bit useless in current form. Can I do this in separate PR?